File: Enzo.Music/Modi/ModoBase.cs
using System;
using System.Collections;
using System.Collections.ObjectModel;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Enzo.Music;

public abstract class ModoBase
{
    protected ReadOnlyCollection<Distanza>? successioni;
    public ReadOnlyCollection<Distanza>? Successioni { get => successioni; }

    public int NumeroSuccessioni => Successioni?.Count ?? 0;

    protected void CreaSuccessioneDistanze(IList<Distanza> lista)
    {
        successioni = new ReadOnlyCollection<Distanza>(lista);
    }
}

File: KantaroBLL/Modi/ModoBase.vb
Public MustInherit Class ModoBase

    Protected Sub CreaSuccessioneDistanze(lista As Generic.IList(Of Distanza))
        _successioni = New System.Collections.ObjectModel.ReadOnlyCollection(Of Distanza)(lista)
    End Sub

    Protected _successioni As System.Collections.ObjectModel.ReadOnlyCollection(Of Distanza)
    Public ReadOnly Property Successioni As System.Collections.ObjectModel.ReadOnlyCollection(Of Distanza)
        Get
            Return _successioni
        End Get
    End Property

    Public ReadOnly Property NumeroSuccessioni As Integer
        Get
            Return Successioni.Count
        End Get
    End Property
End Class
